### PR TITLE
AArch64 Support

### PR DESCRIPTION
**Pull request by AndrewFasano**
Support for Aarch64 (64-bit arm)

Current status:
- [x] Core PANDA/libpanda successfully builds
- [x] Pypanda support & generic qcow
- [x] Syscalls2 support
- [x] OSI support
- [x] All plugins build
- [ ] Record & Replay
- [ ] Testing

Doesn't have support for panda_current_asid, and a bunch of callbacks might be missing

### CODE REVIEW

**Comment by tnballo** [comment]
AAarch64 `syscalls2` should be working now, can get reasonable output:

```
Loading libpanda from /usr/local/lib/python3.6/dist-packages/pandare/data
[PYPANDA] Panda args: [/usr/local/lib/python3.6/dist-packages/pandare/data/aarch64-softmmu/libpanda-aarch64.so -L /usr/local/lib/python3.6/dist-packages/pandare/data/pc-bios -M virt -cpu cortex-a57 -drive file=flash0.qcow,if=pflash,readonly=on ./ubuntu20_04-aarch64.qcow -m 1G -serial unix:/tmp/pypanda_sd_6h4ne0,server,nowait -monitor unix:/tmp/pypanda_mylnpynso,server,nowait]
PANDA[core]:os_familyno=2 bits=64 os_details=ubuntu-1.3.3.7
PANDA[core]:loading required plugin syscalls2
PANDA[core]:initializing syscalls2
PANDA[syscalls2]:using profile for linux aarch64
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/proc/229/cgroup'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering faccessat, char *fname arg = '/usr/bin/uname'
Entering faccessat, char *fname arg = '/usr/bin/uname'
Entering faccessat, char *fname arg = '/usr/bin/uname'
Entering faccessat, char *fname arg = '/usr/bin/uname'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering openat, char *fname arg = '/run/systemd/netif/links/2'
Entering faccessat, char *fname arg = '/etc/ld.so.preload'
Entering openat, char *fname arg = '/etc/ld.so.cache'
Entering openat, char *fname arg = '/lib/aarch64-linux-gnu/libc.so.6'
Entering openat, char *fname arg = '/proc/294/cgroup'
```

File: panda/include/panda/common.h [hunk 1]
```diff
@@ -151,6 +151,7 @@ static inline int panda_virtual_memory_rw(CPUState *env, target_ulong addr,
         // No physical page mapped, even after potential privileged switch, abort
         if (phys_addr == -1)  {
             if (changed_priv) exit_priv(env); // Cleanup mode if necessary
+            printf("no physical page mapped\n");
```
Comment (AndrewFasano) [inline panda/include/panda/common.h:154]: Delete

File: panda/include/panda/common.h [hunk 2]
```diff
@@ -169,6 +170,7 @@ static inline int panda_virtual_memory_rw(CPUState *env, target_ulong addr,
         // Still failed, even after potential privileged switch, abort
         if (ret != MEMTX_OK) {
             if (changed_priv) exit_priv(env); // Cleanup mode if necessary
+            printf("failed after privilege switch\n");
```
Comment (AndrewFasano) [inline panda/include/panda/common.h:173]: Delete

File: panda/plugins/config.panda [hunk 3]
```diff
@@ -28,7 +28,7 @@ osi_test
 pri
 pri_dwarf
 pri_simple
-recctrl
+#recctrl
```
Comment (AndrewFasano) [inline panda/plugins/config.panda:31]: Is this only broken for AArch64? If so let's fix it

**Review by AndrewFasano — COMMENTED** [review]
Looks pretty good to me!

File: panda/src/common.c [hunk 4]
```diff
@@ -98,14 +98,18 @@ target_ulong panda_current_asid(CPUState *cpu) {
 #if defined(TARGET_I386)
   CPUArchState *env = (CPUArchState *)cpu->env_ptr;
   return env->cr[3];
-#elif defined(TARGET_ARM) && !defined(TARGET_AARCH64)
+#elif defined(TARGET_ARM)
+#if defined(TARGET_AARCH64)
+  return 0; // XXX: TODO
```
Comment (AndrewFasano) [inline panda/src/common.c:103]: Let's fix this

**Pull request merged**